crates/aggchain-proof-builder/src/lib.rs

```diff
@@ -282,23 +291,29 @@ impl<ContractsClient> AggchainProofBuilder<ContractsClient> {
         };
 
         {
-            let retrieved_from_contracts = AggregationProofPublicValues::from(&fep_inputs);
-
-            info!(
-                "Aggregation proof public values received with the proof: {:?}",
-                request.aggregation_proof_public_values
-            );
-
-            info!(
-                "Aggregation proof public values retrieved from contracts: {:?}",
-                retrieved_from_contracts
-            );
-
-            if request.aggregation_proof_public_values != retrieved_from_contracts {
-                return Err(Error::MismatchAggregationProofPublicValues {
-                    expected_by_contract: Box::new(retrieved_from_contracts),
-                    expected_by_verifier: Box::new(request.aggregation_proof_public_values),
-                });
+            if let AggchainProofMode::Proof {
+                ref aggregation_proof_public_values,
+                ..
+            } = request.aggchain_proof_mode
+            {
+                let retrieved_from_contracts = AggregationProofPublicValues::from(&fep_inputs);
+
+                info!(
+                    "Aggregation proof public values received with the proof: {:?}",
+                    aggregation_proof_public_values
+                );
+
+                info!(
+                    "Aggregation proof public values retrieved from contracts: {:?}",
+                    retrieved_from_contracts
+                );
+
+                if aggregation_proof_public_values != &retrieved_from_contracts {
+                    return Err(Error::MismatchAggregationProofPublicValues {
```

Comment:
On rust proposer branch, this `info` printouts are moved to error print within the if block. You can leave it this way, we will change it with the merge of the rust proposer branch.

Reply:
I think it's very valuable to keep printing them in all cases, even if no mismatch here, because there is still room for mismatch with the L1 much later in the flow via the `aggchain_hash`